assets/src/edit-story/app/media/utils/createResource.js

```diff
@@ -88,6 +88,7 @@ import getTypeFromMime from './getTypeFromMime';
  * @property {number} width The natural resource width.
  * @property {number} height The natural resource height.
  * @property {string|null} poster The poster URL for the "video" type.
+ * @property {File|null} posterFile The poster file object for the "video" type.
```

Comment:
I would like to avoid adding a new `posterFile` property for this. Any chance we could just re-purpose `poster` for this?

`@property {string|File|null} poster The poster URL or File object for the "video" type.`

Reply:
I don't think this is a good idea. The poster attribute is either, null, blob url or url. If it becomes file object, it may mean we have to add lots of checks everywhere for. The issue, is we need the file object in this single context. 

What would be cleaner, would be removing poster generation out of `getVideoResource`. Maybe something like this. 

```
const getVideoResource = async (file, attrs: {} ) => {
  const fileName = getFileName(file);
  const mimeType = file.type;

  const reader = await createFileReader(file);

  const src = createBlob(new Blob([reader.result], { type: mimeType }));

  const videoEl = document.createElement('video');
  const canPlayVideo = '' !== videoEl.canPlayType(mimeType);

  if( ! attrs.poster ||   ! attrs.width ||  ! attrs.height ) {
       const frame = await getFirstFrameOfVideo(src);

       const poster = createBlob(frame);
       const { width, height } = await getImageDimensions(poster);
       attr = {
          ...attr,
          width, 
          height,
          poster 
       }
  }
  
   return createLocalResource({
    type: 'video',
    mimeType,
    src: canPlayVideo ? src : '',
    alt: fileName,
    title: fileName,
    ...attr,
  });

```

Reply:
Good point regarding adding checks everywhere.

And interesting suggestion to move poster generation out of `getVideoResource()`. Although it could be moved completely, not just wrapped in an `if()`.

While this current usage causes some duplication in calling `getFirstFrameOfVideo()`, it is a bit of a tradeoff to improve UX, as described here:

https://github.com/google/web-stories-wp/blob/92c63e38058fb055c079d70081a86cc023d3dd15/assets/src/edit-story/app/media/useUploadMedia.js#L227-L232

And I think this tradeoff (and thus minor code duplication) is still acceptable.

The major concern is step (8) mentioned in #6904, where we call `uploadVideoFrame()` (and thus `getFirstFrameOfVideo()`) _again_, even though we already have a poster.

I'll try to go over the PR once more to get a better sense of this though.

Reply:
I am sorry this comment is a little unclear to me. Should we just call `getFirstFrameOfVideo` twice here?